support/reattach-pv/main.go

```diff
@@ -207,13 +217,42 @@ type MatchingVolumeClaim struct {
 }
 
 // matchPVsWithClaim iterates over existing pvs to match them to an expected pvc.
-func matchPVsWithClaim(pvs []v1.PersistentVolume, claims map[types.NamespacedName]v1.PersistentVolumeClaim) ([]MatchingVolumeClaim, error) {
+func matchPVsWithClaim(pvs []v1.PersistentVolume, claims map[types.NamespacedName]v1.PersistentVolumeClaim, es esv1.Elasticsearch, oldESName string) ([]MatchingVolumeClaim, error) {
+	// The following regex defines the expected persistent volume claim name format.
+	// The regex patterns for both EsName, and NodeSetName groups are directly from apimachinery validation:
+	// https://github.com/kubernetes/apimachinery/blob/6809593a70331498e028baf9baa3b14b10c75214/pkg/util/validation/validation.go#L178
+	var (
+		r = regexp.MustCompile(fmt.Sprintf(`^elasticsearch-data-(%s)-es-([a-z0-9]([-a-z0-9]*[a-z0-9])?)-([0-9])+$`, oldESName))
```

Comment:
I don't really understand why this is `([a-z0-9]([-a-z0-9]*[a-z0-9])?)` and not just `([a-z0-9][-a-z0-9]*[a-z0-9])` to capture the nodesetName?

I like to use named capturing group to auto-document regexp: `^elasticsearch-data-(?P<clusterName>%s)-es-(?P<nodesetName>[a-z0-9][-a-z0-9]*[a-z0-9])-(?P<ordinal>[0-9])+$`.

Reply:
@thbkrkr I tested both, and you're right.  I initially thought something like `mycluster-thing-name01` would require the trailing `?`, but now I see it does not.

I also removed the named captures, as I wasn't using the names, only the indexes, but I see your point. 

After your below comment about simply replacing the prefix, this whole section is likely going to be removed.